rfcs/text/0040-volume-device.md

```diff
@@ -0,0 +1,138 @@
+# 0000: volume device
+<!-- Leave this ID at 0000. The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC. -->
+
+- Stage: **0 (strawperson)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
+- Date: **TBD** <!-- The ECS team sets this date at merge time. This is the date of the latest stage advancement. -->
+
+<!--
+As you work on your RFC, use the "Stage N" comments to guide you in what you should focus on, for the stage you're targeting.
+Feel free to remove these comments as you go along.
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+This RFC propose adding the volume device fieldset to describe volume storage devices such as hard disks, removable USB, mountable virtual disks such as ISO.
+
+ * volume.mount_name
+ * volume.device_name
+ * volume.dos_name
+ * volume.nt_name
+ * volume.bus_type
+ * volume.writable
+ * volume.default_access
+ * volume.file_system_type
+ * volume.product_id
+ * volume.product_name
+ * volume.vendor_id
+ * volume.vendor_name
+ * volume.serial_number
+ * volume.volume_device_type
+ * volume.action
```

Comment:
`volume.action` will be like "attach", "detach", "mount", "unmount". I'm hesitating to place it here. Seems `event.action` is a better place to save such information. Because these volumes fields will be part of a device event, similar to process events whose `event.action` can be "start"/"end".
thoughts?

Reply:
I'd also prefer to use the existing `event.action` field over introducing an additional `*.action` field, if we can avoid it.